provider/aws/aws.go

```diff
@@ -940,23 +964,31 @@ func groupChangesByNameAndOwnershipRelation(cs Route53Changes) map[string]Route5
 	return changesByOwnership
 }
 
-func (p *AWSProvider) tagsForZone(ctx context.Context, zoneID string, profile string) (map[string]string, error) {
+func (p *AWSProvider) tagsForZone(ctx context.Context, zoneIDs []string, profile string) (zoneTags, error) {
 	client := p.clients[profile]
 
-	// TODO: this will make single API request for each zone, which consumes API requests
-	// more effective way is to batch requests https://github.com/aws/aws-sdk-go-v2/blob/ed8a3caa0df9ce36a5b60aebeee201187098d205/service/route53/api_op_ListTagsForResources.go#L37
-	response, err := client.ListTagsForResource(ctx, &route53.ListTagsForResourceInput{
-		ResourceType: route53types.TagResourceTypeHostedzone,
-		ResourceId:   aws.String(cleanZoneID(zoneID)),
-	})
-	if err != nil {
-		return nil, provider.NewSoftErrorf("failed to list tags for zone %s: %w", zoneID, err)
-	}
-	tagMap := map[string]string{}
-	for _, tag := range response.ResourceTagSet.Tags {
-		tagMap[*tag.Key] = *tag.Value
+	result := zoneTags{}
+	// Currently supported up to 10 health checks or hosted zones.
```

Comment:
Where does this 10 comes from ?

Reply:
updated, created a constant with the link to docs